examples/components/CCFRAUD/preprocessing/run.py

```diff
@@ -0,0 +1,270 @@
+        "is_fraud",
+    ]
+    categorical = ["category", "region", "gender", "state", "job"]
+    datetimes = ["trans_date_trans_time"]  # "dob"
+    normalize = [
+        "age",
+        "merch_lat",
+        "merch_long",
+        "lat",
+        "long",
+        "city_pop",
+        "trans_date_trans_time",
+        "amt",
+    ]
+
+    df.loc[:, "age"] = (pd.Timestamp.now() - pd.to_datetime(df["dob"])) // pd.Timedelta(
+        "1y"
+    )
+
+    # Filter only useful columns
+    df = df[useful_props]
+
+    for column in categorical:
+        if column not in ENCODERS:
+            print(f"Creating encoder for column: {column}")
+            # Simply set all zeros if the category is unseen
+            encoder = OneHotEncoder(handle_unknown="ignore")
+            encoder.fit(df[column].values.reshape(-1, 1))
+            ENCODERS[column] = encoder
+
+        encoder = ENCODERS.get(column)
+        encoded_data = encoder.transform(df[column].values.reshape(-1, 1)).toarray()
+        encoded_df = pd.DataFrame(
+            encoded_data,
+            columns=[
+                column + "_" + "_".join(x.split("_")[1:])
+                for x in encoder.get_feature_names()
+            ],
+        )
+        encoded_df.index = df.index
+        df = df.join(encoded_df).drop(column, axis=1)
+
+    for column in datetimes:
+        df.loc[:, column] = pd.to_datetime(df[column]).view("int64")
+    for column in normalize:
+        if column not in SCALERS:
+            print(f"Creating encoder for column: {column}")
+            # Simply set all zeros if the category is unseen
+            scaler = StandardScaler()
+            scaler.fit(df[column].values.reshape(-1, 1))
+            SCALERS[column] = scaler
+
+        scaler = SCALERS.get(column)
+        df.loc[:, column] = scaler.transform(df[column].values.reshape(-1, 1))
+
+    return df
+
+
+def preprocess_data(
+    raw_training_data,
+    raw_testing_data,
+    train_data_dir="./",
+    test_data_dir="./",
+    metrics_prefix="default-prefix",
+    config=None,
+):
+    """Preprocess the raw_training_data and raw_testing_data and save the processed data to train_data_dir and test_data_dir.
+
+    Args:
+        raw_training_data: Training data directory that need to be processed
+        raw_testing_data: Testing data directory that need to be processed
+        train_data_dir: Train data directory where processed train data will be saved
+        test_data_dir: Test data directory where processed test data will be saved
+    Returns:
+        None
+    """
+
+    logger.info(
+        f"Raw Training Data path: {raw_training_data}, Raw Testing Data path: {raw_testing_data}, Processed Training Data dir path: {train_data_dir}, Processed Testing Data dir path: {test_data_dir}"
+    )
+
+    logger.debug('Loading regions from "us_regions.csv"')
+    global STATES_REGIONS
+    df_states_regions = pd.read_csv(hydra.utils.get_original_cwd() + "/us_regions.csv")
```

Comment:
Actually, for this one you have to use hydra because using hydra sets os cwd to its own output folder and thus in order to get original execution cwd one need to use hydra api